tests/unit-tests/test-class-usage-tracking.php

```diff
@@ -0,0 +1,318 @@
+<?php
+
+class Sensei_Usage_Tracking_Test extends WP_UnitTestCase {
+	public function setup() {
+		parent::setup();
+		$this->factory = new Sensei_Factory();
+		$this->usage_tracking = new Sensei_Usage_Tracking();
+	}
+
+	public function teardown() {
+		parent::teardown();
+	}
+
+	/**
+	 * Ensure cron job action is set up.
+	 */
+	public function test_cron_job_action_added() {
+		$this->usage_tracking->hook();
+		$this->assertTrue( !! has_action( 'sensei_core_jobs_usage_tracking_send_data', array( $this->usage_tracking, 'maybe_send_usage_data' ) ) );
+	}
+
+	/**
+	 * Ensure scheduling function works properly.
+	 */
+	public function test_maybe_schedule_tracking_task() {
+		// Make sure it's cleared initially
+		wp_clear_scheduled_hook( 'sensei_core_jobs_usage_tracking_send_data' );
+
+		// Record how many times the event is scheduled
+		$event_count = 0;
+		add_filter( 'schedule_event', function( $event ) use ( &$event_count ) {
+			if ( $event->hook === 'sensei_core_jobs_usage_tracking_send_data' ) {
+				$event_count++;
+			}
+			return $event;
+		} );
+
+		// Should successfully schedule the task
+		$this->assertFalse( wp_get_schedule( 'sensei_core_jobs_usage_tracking_send_data' ) );
+		$this->usage_tracking->maybe_schedule_tracking_task();
+		$this->assertNotFalse( wp_get_schedule( 'sensei_core_jobs_usage_tracking_send_data' ) );
+		$this->assertEquals( 1, $event_count );
```

Comment:
An alternative to this may be to use `wp_next_scheduled` and then just compare return values between calls to ensure the timestamp is the same. It would eliminate the need for this bit - https://github.com/Automattic/sensei/pull/1984/files#diff-0ea0511e3f117834481aeca98312467fR31.

Reply:
I'd love to eliminate the filter. The problem with using `wp_next_scheduled`, however, is that in this case the "next scheduled" job is always going to be the first one that was added, regardless of whether more were scheduled after it. So I can't use it to tell whether additional jobs were scheduled, as far as I know.

Reply:
Oh, that's too bad. 🙁